Version/Version.cpp

```diff
@@ -0,0 +1,96 @@
+#include <filesystem>
+#include <Windows.h>
+#include <iostream>
+
+std::filesystem::path get_exe_path()
+{
+    TCHAR buffer[MAX_PATH];
+    GetModuleFileNameW(NULL, buffer, _countof(buffer));
+    return std::filesystem::path(buffer).parent_path();
+}
+
+std::filesystem::path find_api_directory()
+{
+    return get_exe_path()/(TEXT(R"(ArkApi)"));
+}
+
+std::filesystem::path find_dll()
+{
+    return find_api_directory()/(TEXT(R"(AsaApi.dll)"));
+}
+
+std::filesystem::path get_sys_path()
+{
+	TCHAR sys_dir[MAX_PATH];
+	GetSystemDirectoryW(sys_dir, MAX_PATH);
+	return std::filesystem::path(sys_dir);
+}
+
+std::filesystem::path find_vdll()
+{
+	return get_sys_path()/(TEXT(R"(Version.dll)"));
```

Comment:
`TEXT` is a macro that simply prepends L when UNICODE is defined.

Honestly, the filesystem api is smart enough to automagically handle either, this can just be
```cpp
return gey_sys_path()/"version.dll";
```